backend/middleware/authMiddleware.js

```diff
@@ -0,0 +1,46 @@
+const asyncHandler = require("express-async-handler")
+const User = require("../model/user.model")
+const jwt = require("jsonwebtoken")
+const { ErrorHandler } = require("./errorMiddleware")
+
+const protect = asyncHandler(async (req, res, next) => {
```

Comment:
Can we merge protect and authorize middleware? It seems both are having same job i.e. to check whether function is allowed to access or not.

Reply:
Done